pygdf/dataframe.py

```diff
@@ -353,6 +384,8 @@ def add_column(self, name, data, forceindex=False):
         data : Series, array-like
             Values to be added.
         """
+        data = self._sanitize_columns(data)
```

Comment:
I think it's better to move the call to `_sanitize_columns` into `_prepare_series_for_add`, which already contains some check/sanitize logic.